kernel/mmanutils.c

```diff
@@ -406,8 +406,10 @@ long myst_mmap(
         // ATTN: Use EBADF and EACCES for fd validation failures. This may not
         // conform the Linux kernel behavior.
 
-        /* fail if not a regular file */
-        if (myst_syscall_fstat(fd, &buf) != 0 || !S_ISREG(buf.st_mode))
+        /* fail if not a regular file or character device(some apps map
+         * /dev/zero) */
+        if (myst_syscall_fstat(fd, &buf) != 0 ||
```

Comment:
Why is that the failure is inconsistent? My understanding is that it should have been consistent if this was the fix.

Reply:
This is a recent regression. We had a [commit](https://github.com/deislabs/mystikos/commit/4f4d1e563fa89fca749ce7ce257c98b360f0a626) last week to change the file mode on /dev/zero. 
I would imagine JIT/IL_Conformance/Old/Base/add_ovf/add_ovf.dll coreclr test has been failing consistently since.